proxy/reverse.go

```diff
@@ -134,6 +135,11 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		}
 		redirectRequest(proxyreq, ep.URL)
 
+		if repeated := p.checkProxyLoop(proxyreq, ep.URL.String()); repeated {
```

Comment:
could we move this check to the beginning of `ServeHTTP`? If it cannot pass the check, there is no need to try to proxy this request. It makes the logic more straightforward.